Week4/Solution.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Alyssa Lingad
+ * 
+ * MICP: Week 4
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.*;
+
+
+
+public class Solution
+{
+	public static boolean canBeSegmented(String str, HashSet d) throws Exception{
+		int i = 0;
+		if (str == null)
+			throw new Exception();
+		if (str.isEmpty())
+			return false;
+		if (d.isEmpty())
+			return false;
+		
+		int size = str.length();
+		String s = str.toLowerCase();
+		boolean check = false;
+		while (i < size){
+			int j = i + 2;
```

Comment:
Is there a reason why j is set to i+2? I mean shouldn't this be i+1 or i+0 if you are considering to find the minimum sub-string and then check if that is a string that exists in the dictionary or not?